Framework/API/test/WorkspacePropertyWithIndexHelperTest.h

```diff
@@ -0,0 +1,34 @@
+#ifndef MANTID_API_WORKSPACEWITHINDEXHELPERTEST_H_
+#define MANTID_API_WORKSPACEWITHINDEXHELPERTEST_H_
+
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/WorkspacePropertyWithIndexHelper.h"
+#include <cxxtest/TestSuite.h>
+
+using namespace Mantid::API;
+
+class WorkspaceWithIndexHelperTest : public CxxTest::TestSuite {
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static WorkspaceWithIndexHelperTest *createSuite() {
+    return new WorkspaceWithIndexHelperTest();
+  }
+  static void destroySuite(WorkspaceWithIndexHelperTest *suite) {
+    delete suite;
+  }
+
+  void testCreateWithSpectrumNumbers() {
+    TS_ASSERT_THROWS_NOTHING(createWithSpectrumNumbers<MatrixWorkspace>());
```

Comment:
Yes this needs to be removed altogether. The builder functions aren't necessary.